doc/contributing.md

```diff
@@ -0,0 +1,128 @@
+# Contributing to Emboss
+
+If you would like to fix a bug or add a new feature to Emboss, great!  This
+document is intended to help you understand the procedure, so that your change
+can land in the main Emboss repository.
+
+You do not have to take a change all the way from start to finish, either: if
+you can get a design approved, then someone else can implement it much more
+easily.  Conversely, if you are looking for a way to help, you might look for
+existing [feature
+requests](https://github.com/google/emboss/labels/enhancement) that have
+designs or at [open design sketches](design_docs/) that you might be able to
+implement.
+
+
+## All Changes
+
+Because Emboss is a Google project, in order to submit code you will need to
+sign a [Google Contributor License Agreement
+(CLA)](https://cla.developers.google.com/).
+
+**IMPORTANT**: if your contribution includes code that is not covered by a
+Google CLA and is not owned by Google, the Emboss project has to follow special
+procedures to include it.  Please let us know ([filing an issue on
+GitHub](https://github.com/google/emboss/issues/new) is probably the easiest
+way) so that we can walk you through the process.  In particular, we generally
+cannot accept any code from StackExchange or similar sites, and any code that
+comes from a non-Google open source project needs to have an acceptable license
+and be committed to the Emboss repository in a specific location.
+
+
+### How-To Guides
+
+This document covers the process of getting a change into the main Emboss
+repository — i.e., what you need to do to get your change into
+[the main Emboss repository](https://github.com/google/emboss/).
+
+[How to Implement Changes to Emboss](how-to-implement.md) has an overview of
+how to make code changes to Emboss.
+
+[How to Design Features for Emboss](how-to-design.md) has an overview of what
+to think about during your design.
+
+
+### Bug Fixes vs New Features
+
+The general process for bug fixes and new features is the same, but bug fixes
+usually require less design work, and therefore can go through lighter
+processes.
+
+
+## Very Small Changes
+
+If you have a tiny change — for example,  making a fix that does not change the
+design of `embossc` — you can jump directly into coding.
+
+This process works best if your change is small and not likely to be
+controversial, or if you have already completed [the steps for small
+changes](#small-changes).
+
+1.  [Code up your proposed change](how-to-implement.md) and open a [pull
+    request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request)
+    to [the main Emboss repository](https://github.com/google/emboss/).  If you
+    have not completed [the steps for small changes](#small-changes), this
+    gives the Emboss maintainers something more concrete to look at, but you
+    may end up doing more work if your initial proposal is rejected.
```

Comment:
We could maybe use a different term that "rejected" here (point 2 below does a good job framing this)

Reply:
Done.